apps/react-app/src/pods/template-export/template-export.component.tsx

```diff
@@ -1,7 +1,12 @@
 import React from 'react';
 
+import { MacContext } from '@/providers/mac.context';
+
 export const TemplateExport: React.FC = () => {
 
+  const {macData} = React.useContext(MacContext);
```

Comment:
This should be in the container, the dumb component (presentational component) will hold the export button, and bubble up to the container what to do.

This approach is not written in stone, for this case makes sense because there's no deep hierarchy of components, in other cases it may vary (usually, container / component, if you go deep with components, then think about UsecContext, if things gets complex think about useReducer, but this can be incorporated whenever is needed or if you realize it will happen soon)